### PR TITLE
docs(components): [api-external]

### PR DESCRIPTION
**Pull request by chenxch**
- Add external API

before:
image

```
text, textarea and other [native input types](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types)
```

after:
image

```
<ExternalType prefix="text, textarea and other" text="native input types" url="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types" />
```
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: docs/.vitepress/vitepress/components/globals/vp-api-external.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      viewBox="0 0 24 24"
+      width="1.2em"
+      height="1.2em"
+      class="link-icon"
+    >
+      <path
+        fill="currentColor"
+        d="M10 6v2H5v11h11v-5h2v6a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1V7a1 1 0 0 1 1-1h6zm11-3v8h-2V6.413l-7.793 7.794l-1.414-1.414L17.585 5H13V3h8z"
+      />
+    </svg>
```
Comment (jw-foss) [inline docs/.vitepress/vitepress/components/globals/vp-api-external.vue:33]: Actually, we have this icon in the code base, you may refer to file `docs/.vitepress/vitepress/components/common/vp-link.vue`

**Review by jw-foss — COMMENTED** [review]
@chenxch left a comment for you, BTW, would you mind putting a screenshot in the PR descriptions to make it more comprehensive? Thanks!

**Comment by chenxch** [comment]
@jw-foss PTAL.

**Comment by jw-foss** [comment]
@chenxch please revert changes in `input.md`, we should do that in a dedicated PR.

**Comment by chenxch** [comment]
> @chenxch please revert changes in `input.md`, we should do that in a dedicated PR.

You're right.

**Comment by jw-foss** [comment]
@chenxch I got a question for you, why were we having this empty file named `api-deprecated.vue`, if we are not writing it in this PR, we should delete it, otherwise I think you should implement it 🤔

**Comment by chenxch** [comment]
> @chenxch I got a question for you, why were we having this empty file named `api-deprecated.vue`, if we are not writing it in this PR, we should delete it, otherwise I think you should implement it 🤔

I'm so sorry I didn't remove it when I found out and similar components existed, it was my mistake.

**Review by jw-foss — APPROVED** [review]

**Pull request merged**